packages/test/test-end-to-end-tests/src/test/handleValidation.spec.ts

```diff
@@ -0,0 +1,350 @@
+			return dirRoot.get("B") as IFluidHandle<ITestFluidObject>;
+		}
+	},
+	{
+		type: stringId,
+		storeHandle: async (defaultDataStore, handle) => {
+			const stringRoot = await defaultDataStore.getSharedObject(stringId);
+			stringRoot.annotateRange(0, 1, { B: handle });
+		},
+		readHandle: async (defaultDataStore) => {
+			const stringRoot = await defaultDataStore.getSharedObject(stringId);
+			return stringRoot.getPropertiesAtPosition(0)?.B as IFluidHandle<ITestFluidObject>;
+		}
+	},
+	{
+		type: matrixId,
+		storeHandle: async (defaultDataStore, handle) => {
+			const matrixRoot = await defaultDataStore.getSharedObject(matrixId);
+			matrixRoot.insertRows(0, 1);
+			matrixRoot.insertCols(0, 1);
+			matrixRoot.setCell(0, 0, handle);
+		},
+		readHandle: async (defaultDataStore) => {
+			const matrixRoot = await defaultDataStore.getSharedObject(matrixId);
+			return matrixRoot.getCell(0, 0) as IFluidHandle<ITestFluidObject>;
+		}
+	},
+	{
+		type: treeId,
+		storeHandle: async (defaultDataStore, handle) => {
+			const treeRoot = await defaultDataStore.getSharedObject(treeId);
+
+			const treeView = treeSetup(treeRoot);
+			treeView.root.h = handle;
+		},
+		readHandle: async (defaultDataStore) => {
+			const treeRoot = await defaultDataStore.getSharedObject(treeId);
+			const treeView = treeSetup(treeRoot);
+			return treeView.root.h as IFluidHandle<ITestFluidObject>;
+		}
+	},
+	// {
+	// 	type: legacyTreeId,
+	// 	storeHandle: async (defaultDataStore, handle) => {
+	// 		const treeRoot = await defaultDataStore.getSharedObject(legacyTreeId);
+	// 		const legacyNodeId: TraitLabel = "inventory" as TraitLabel;
+
+	// 		const handleNode: BuildNode = {
+	// 			definition: legacyNodeId,
+	// 			traits: {
+	// 				handle,
+	// 			},
+	// 		};
+	// 		treeRoot.applyEdit(
+	// 			Change.insertTree(
+	// 				handleNode,
+	// 				StablePlace.atStartOf({
+	// 					parent: treeRoot.currentView.root,
+	// 					label: legacyNodeId,
+	// 				}),
+	// 			),
+	// 		);
+
+	// 		const rootNode = treeRoot.currentView.getViewNode(treeRoot.currentView.root);
+	// 		// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+	// 		const nodeId = rootNode.traits.get(legacyNodeId)![0];
+	// 		const change: Change = Change.setPayload(nodeId, handle);
+	// 		treeRoot.applyEdit(change);
+	// 	},
+	// },
+	{
+		type: registerId,
+		storeHandle: async (defaultDataStore, handle) => {
+			const registerRoot = await defaultDataStore.getSharedObject(registerId);
+			registerRoot.write("B", handle);
+		},
+		readHandle: async (defaultDataStore) => {
+			const registerRoot = await defaultDataStore.getSharedObject(registerId);
+			return registerRoot.read("B") as IFluidHandle<ITestFluidObject>;
+		}
+	},
+	{
+		type: queueId,
+		storeHandle: async (defaultDataStore, handle) => {
+			const queueRoot = await defaultDataStore.getSharedObject(queueId);
+			queueRoot.add(handle);
+		},
+		readHandle: async (defaultDataStore) => {
```

Comment:
I know this is not correct - how exactly can I get the value out of ConsensusQueue when it's not returned directly? Acquire returns a boolean based on whether or not there are items in the queue, and I couldn't find another get-like method

Reply:
this currently looks right. i think there is also and aquire that waits for the value that we probably want here so it works on real servers